Cannot drop db_discovery type

## Bug Report

### Which version of ShardingSphere did you use?
62005f74f508ff0e5ae3d5882fbbf0cd19c4784f
### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
Proxy
### Expected behavior
db_discovery type dropped
### Actual behavior

when execute `drop db_discovery rule` exception cannot found db_discovery type

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.
1.  Register storage units
```sql
register storage unit 
ds_0(url="jdbc:mysql://xxx.xxx.xxx.xxx:3306/primary_demo_ds?serverTimezone=UTC&useSSL=false", user="root",password="xxxxx"), 
ds_1(url="jdbc:mysql://xxx.xxx.xxx.xxx:3307/primary_demo_ds?serverTimezone=UTC&useSSL=false", user="root",password="xxxxx"),
ds_2(url="jdbc:mysql://xxx.xxx.xxx.xxx:3308/primary_demo_ds?serverTimezone=UTC&useSSL=false", user="root",password="xxxxx");
```
2. Create db_discovery rule
```sql
CREATE DB_DISCOVERY RULE group_0 ( STORAGE_UNITS(ds_0, ds_1, ds_2), TYPE(NAME='MySQL.MGR',PROPERTIES('group-name'='666edd3c-02ec-11ea-9bb3-000027e33bd2')), HEARTBEAT(PROPERTIES('keep-alive-cron'='0/5 * * * * ?')) );
```
3. Drop db_discovery type
```sql
DROP DB_DISCOVERY TYPE group_0_MySQL.MGR FROM test0;
```
### Screen shot
<img width="749" alt="image" src="https://user-images.githubusercontent.com/57847965/198498682-d107a50e-0ae7-46d4-a5d1-2dd9076cf8e8.png">

### Log
```
[ERROR] 2022-10-28 11:27:30.135 [Curator-SafeNotifyService-0] c.g.common.eventbus.EventBus.default - Exception thrown by subscriber method renew(org.apache.shardingsphere.mode.manager.cluster.coordinator.registry.status.storage.event.StorageNodeChangedEvent) on subscriber org.apache.shardingsphere.mode.manager.cluster.coordinator.ClusterContextManagerCoordinator@79b84841 when dispatching event: org.apache.shardingsphere.mode.manager.cluster.coordinator.registry.status.storage.event.StorageNodeChangedEvent@4a15eb17
java.lang.NullPointerException: Can not find database discovery data source rule in database `test0`
```

--- a/features/db-discovery/distsql/handler/src/main/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverter.java
+++ b/features/db-discovery/distsql/handler/src/main/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverter.java
@@ -69,7 +69,7 @@ public final class DatabaseDiscoveryRuleStatementConverter {
     }
     
     private static String getName(final String ruleName, final String type) {
-        return String.format("%s_%s", ruleName, type.replace(".", "_"));
+        return String.format("%s_%s", ruleName, type.replace(".", "_").toLowerCase());
     }
     
     /**

--- a/features/db-discovery/distsql/handler/src/main/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverter.java
+++ b/features/db-discovery/distsql/handler/src/main/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverter.java
@@ -69,7 +69,7 @@ public final class DatabaseDiscoveryRuleStatementConverter {
     }
     
     private static String getName(final String ruleName, final String type) {
-        return String.format("%s_%s", ruleName, type);
+        return String.format("%s_%s", ruleName, type.replace(".", "_"));
     }
     
     /**

--- a/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverterTest.java
+++ b/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverterTest.java
@@ -39,13 +39,13 @@ public final class DatabaseDiscoveryRuleStatementConverterTest {
     @Test
     public void assertConvert() {
         DatabaseDiscoveryRuleConfiguration ruleConfig = DatabaseDiscoveryRuleStatementConverter.convert(createDatabaseDiscoveryRuleSegments());
-        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("definition_MySQL_MGR"));
+        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("definition_mysql_mgr"));
         assertTrue(ruleConfig.getDiscoveryHeartbeats().containsKey("definition_heartbeat"));
         Iterator<DatabaseDiscoveryDataSourceRuleConfiguration> iterator = ruleConfig.getDataSources().iterator();
         DatabaseDiscoveryDataSourceRuleConfiguration dataSourceRuleConfig = iterator.next();
         assertThat(dataSourceRuleConfig.getDataSourceNames(), is(Arrays.asList("resource0", "resource1")));
         assertThat(dataSourceRuleConfig.getGroupName(), is("definition"));
-        assertThat(dataSourceRuleConfig.getDiscoveryTypeName(), is("definition_MySQL_MGR"));
+        assertThat(dataSourceRuleConfig.getDiscoveryTypeName(), is("definition_mysql_mgr"));
         assertThat(dataSourceRuleConfig.getDiscoveryHeartbeatName(), is("definition_heartbeat"));
     }
     

--- a/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverterTest.java
+++ b/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/converter/DatabaseDiscoveryRuleStatementConverterTest.java
@@ -39,13 +39,13 @@ public final class DatabaseDiscoveryRuleStatementConverterTest {
     @Test
     public void assertConvert() {
         DatabaseDiscoveryRuleConfiguration ruleConfig = DatabaseDiscoveryRuleStatementConverter.convert(createDatabaseDiscoveryRuleSegments());
-        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("definition_MySQL.MGR"));
+        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("definition_MySQL_MGR"));
         assertTrue(ruleConfig.getDiscoveryHeartbeats().containsKey("definition_heartbeat"));
         Iterator<DatabaseDiscoveryDataSourceRuleConfiguration> iterator = ruleConfig.getDataSources().iterator();
         DatabaseDiscoveryDataSourceRuleConfiguration dataSourceRuleConfig = iterator.next();
         assertThat(dataSourceRuleConfig.getDataSourceNames(), is(Arrays.asList("resource0", "resource1")));
         assertThat(dataSourceRuleConfig.getGroupName(), is("definition"));
-        assertThat(dataSourceRuleConfig.getDiscoveryTypeName(), is("definition_MySQL.MGR"));
+        assertThat(dataSourceRuleConfig.getDiscoveryTypeName(), is("definition_MySQL_MGR"));
         assertThat(dataSourceRuleConfig.getDiscoveryHeartbeatName(), is("definition_heartbeat"));
     }
     

--- a/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/AlterDatabaseDiscoveryRuleStatementUpdaterTest.java
+++ b/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/AlterDatabaseDiscoveryRuleStatementUpdaterTest.java
@@ -108,7 +108,7 @@ public final class AlterDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(databaseDiscoveryRuleConfiguration.getDataSources().size(), is(1));
         assertTrue(databaseDiscoveryRuleConfiguration.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(databaseDiscoveryRuleConfiguration.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL.MGR"));
+        assertTrue(databaseDiscoveryRuleConfiguration.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
         assertTrue(databaseDiscoveryRuleConfiguration.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
     
@@ -125,7 +125,7 @@ public final class AlterDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(currentConfig.getDataSources().size(), is(1));
         assertTrue(currentConfig.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL.MGR"));
+        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
         assertTrue(currentConfig.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
 }

--- a/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/AlterDatabaseDiscoveryRuleStatementUpdaterTest.java
+++ b/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/AlterDatabaseDiscoveryRuleStatementUpdaterTest.java
@@ -108,7 +108,7 @@ public final class AlterDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(databaseDiscoveryRuleConfiguration.getDataSources().size(), is(1));
         assertTrue(databaseDiscoveryRuleConfiguration.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(databaseDiscoveryRuleConfiguration.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
+        assertTrue(databaseDiscoveryRuleConfiguration.getDiscoveryTypes().containsKey("readwrite_ds_1_mysql_mgr"));
         assertTrue(databaseDiscoveryRuleConfiguration.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
     
@@ -125,7 +125,7 @@ public final class AlterDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(currentConfig.getDataSources().size(), is(1));
         assertTrue(currentConfig.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
+        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_mysql_mgr"));
         assertTrue(currentConfig.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
 }

--- a/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/CreateDatabaseDiscoveryRuleStatementUpdaterTest.java
+++ b/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/CreateDatabaseDiscoveryRuleStatementUpdaterTest.java
@@ -98,7 +98,7 @@ public final class CreateDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(ruleConfig.getDataSources().size(), is(1));
         assertTrue(ruleConfig.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL.MGR"));
+        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
         assertTrue(ruleConfig.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
     
@@ -113,7 +113,7 @@ public final class CreateDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(currentConfig.getDataSources().size(), is(1));
         assertTrue(currentConfig.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL.MGR"));
+        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
         assertTrue(currentConfig.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
 }

--- a/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/CreateDatabaseDiscoveryRuleStatementUpdaterTest.java
+++ b/features/db-discovery/distsql/handler/src/test/java/org/apache/shardingsphere/dbdiscovery/distsql/handler/update/CreateDatabaseDiscoveryRuleStatementUpdaterTest.java
@@ -98,7 +98,7 @@ public final class CreateDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(ruleConfig.getDataSources().size(), is(1));
         assertTrue(ruleConfig.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
+        assertTrue(ruleConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_mysql_mgr"));
         assertTrue(ruleConfig.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
     
@@ -113,7 +113,7 @@ public final class CreateDatabaseDiscoveryRuleStatementUpdaterTest {
         assertThat(currentConfig.getDataSources().size(), is(1));
         assertTrue(currentConfig.getDataSources().stream().map(DatabaseDiscoveryDataSourceRuleConfiguration::getGroupName)
                 .collect(Collectors.toList()).removeAll(Collections.singletonList("readwrite_ds_1")));
-        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_MySQL_MGR"));
+        assertTrue(currentConfig.getDiscoveryTypes().containsKey("readwrite_ds_1_mysql_mgr"));
         assertTrue(currentConfig.getDiscoveryHeartbeats().containsKey("readwrite_ds_1_heartbeat"));
     }
 }

